src/component/handlers/edit/editOnPaste.js

```diff
@@ -146,9 +146,21 @@ function editOnPaste(e: SyntheticClipboardEvent): void {
         this.props.blockRenderMap
       );
       if (htmlFragment) {
-        var htmlMap = BlockMapBuilder.createFromArray(htmlFragment);
-        this.update(insertFragment(this.props.editorState, htmlMap));
-        return;
+        const { contentBlocks, contentState } = htmlFragment;
```

Comment:
You understood correctly. We could definitely just return an `EntityMap` instance and update `insertFragment` to support that. So we'd do `insertFragment(this.props.editorState, htmlMap, entityMap)`. Sounds good to me!